library/platform_util.c

```diff
@@ -65,3 +74,59 @@ void mbedtls_platform_zeroize( void *buf, size_t len )
     memset_func( buf, 0, len );
 }
 #endif /* MBEDTLS_PLATFORM_ZEROIZE_ALT */
+
+#if defined(MBEDTLS_HAVE_TIME_DATE) && !defined(MBEDTLS_PLATFORM_GMTIME_R_ALT)
+#include <time.h>
+#if !defined(_WIN32) && (defined(unix) || \
+    defined(__unix) || defined(__unix__) || (defined(__APPLE__) && \
+    defined(__MACH__)))
+#include <unistd.h>
+
+#if !( ( defined(_POSIX_VERSION) && _POSIX_VERSION >= 200809L ) ||     \
+       ( defined(_POSIX_THREAD_SAFE_FUNCTIONS ) &&                     \
+         _POSIX_THREAD_SAFE_FUNCTIONS >= 20112L ) )
+/*
+ * This is a convenience shorthand macro to avoid checking the long
+ * preprocessor conditions above. Ideally, we could expose this macro in
+ * platform_util.h and simply use it in platform_util.c, threading.c and
+ * threading.h. However, this macro is not part of the Mbed TLS public API, so
+ * we keep it private by only defining it in this file
+ */
+#define PLATFORM_UTIL_USE_GMTIME
```

Comment:
If one wanted to keep using gmtime() in non-posix OS (such as Mbed OS), how would that happen as this define is inside "#if <some-unix-flavour>" -block starting at line 80?

Reply:
I see your point (finally) - on a non-windows, non-unix system, we don't define `PLATFORM_UTIL_USE_GMTIME`, so we're using `gmtime_r()` by default, which is not what we want.

Reply:
@TeroJaasko I pushed 323d801 to address your point. My apologies for missing it for so long.

Reply:
Thank you for the change. Now it seems that it can work for us and we re-started tests.